tests/FakeItEasy.Specs/EventRaisingSpecs.cs

```diff
@@ -189,136 +204,167 @@ public void NullSenderAndEventArguments()
+            "And the second subscriber is invoked once"
                 .x(() => handler2InvocationCount.Should().Be(1));
         }
 
         [Scenario]
         public void CustomEventArguments()
         {
-            "when raising generic event passing arguments"
+            "Given an event of type EventHandler with custom event arguments type"
+                .See(() => nameof(Fake.GenericEvent));
+
+            "When I raise the event specifying only the event arguments"
                 .x(() => Fake.GenericEvent += Raise.With(this.customEventArgs));
 
-            "it should pass the fake as sender"
+            "Then the fake is passed as the sender"
                 .x(() => CapturedSender.Should().BeSameAs(Fake));
 
-            "it should pass the event arguments"
+            "And the supplied value is passed as the event arguments"
                 .x(() => CapturedArgs1.Should().BeSameAs(this.customEventArgs));
         }
 
         [Scenario]
         public void SenderAndCustomEventArguments()
         {
-            "when raising generic event passing sender and arguments"
+            "Given an event of type EventHandler with custom event arguments type"
+                .See(() => nameof(Fake.GenericEvent));
+
+            "When I raise the event specifying the sender and the event arguments"
                 .x(() => Fake.GenericEvent += Raise.With(SampleSender, this.customEventArgs));
 
-            "it should pass the sender"
+            "Then the supplied sender is passed as the event sender"
                 .x(() => CapturedSender.Should().BeSameAs(SampleSender));
 
-            "it should pass the event arguments"
+            "And the supplied value is passed as the event arguments"
                 .x(() => CapturedArgs1.Should().BeSameAs(this.customEventArgs));
         }
 
         [Scenario]
         public void NullSenderAndCustomEventArguments()
         {
-            "when raising generic event passing arguments and null sender"
+            "Given an event of type EventHandler with custom event arguments type"
+                .See(() => nameof(Fake.GenericEvent));
+
+            "When I raise the event specifying the event arguments and a null sender"
                 .x(() => Fake.GenericEvent += Raise.With(null, this.customEventArgs));
 
-            "it should pass null as the sender"
+            "Then null is passed as the event sender"
                 .x(() => CapturedSender.Should().BeNull());
 
-            "it should pass the event arguments"
+            "And the supplied value is passed as the event arguments"
                 .x(() => CapturedArgs1.Should().BeSameAs(this.customEventArgs));
         }
 
         [Scenario]
         public void CustomEventHandler()
         {
-            "when raising custom event passing sender and arguments"
+            "Given an event of type EventHandler with custom event arguments type"
+                .See(() => nameof(Fake.GenericEvent));

```

Comment:
Whoops. So it should. Thanks! I think the test should appear below the new one as well.